crates/db-allocator/src/lib.rs

```diff
@@ -48,35 +61,89 @@ impl Constraint {
     }
 
     /// Set the min constraint.
-    pub fn min<T>(mut self, min: T) -> Self
+    pub fn min<T>(&mut self, min: T) -> Result<()>
```

Comment:
It would be more nature to introduce a method "Constraint::validate()" instead of returning error from accessor methods.

Reply:
sure, let me redesign this part

Reply:
changed. Constraint::validate() is added and will be used in Interval Tree in later pull request.